### ISSUE TITLE
Publish on nuget.org

### ISSUE THREAD

**Issue by ismcagdas** [loic-sharma/BaGet#262]
First of all, thank you for such a great project :).

### Describe the solution you'd like

Are you willing to publish BaGet as a nuget package on nuget.org ? We want to use it as a NuGet package in our project. There is a nuget package on https://www.nuget.org/packages/BaGet but it seems like you have unlisted it.

Thanks.

**Comment by loic-sharma** [comment]
> We want to use it as a NuGet package in our project.

What would you like to do exactly? You may not need the full BaGet project depending on your scenario.

> Are you willing to publish BaGet as a nuget package on nuget.org ? ... There is a nuget package on https://www.nuget.org/packages/BaGet but it seems like you have unlisted it.

I'm working on bringing this back! I'd like to turn BaGet into a global tool:

```
dotnet tool install -g baget
baget --option1 value1
```

You can track my progress here on the [`feature/global-tool` branch](https://github.com/loic-sharma/BaGet/tree/feature/global-tool).

> First of all, thank you for such a great project :).

Thank you for the kind words! I've put in way too many hours into this project 😅

**Comment by ismcagdas** [comment]
We want to host a private NuGet server and do the followings in it:

* Authorize clients with a license code or something similar in the url. We are planning to do this using ASP.NET Core's action filters. In that way, we don't need to modify BaGet's source code to do that. 

For example each separate user/customer will use something like http://mywebsite.com/license-code/packages/*** or http://mywebsite.com/packages/***?licenseCode={license-code}

* Our second requirement is filter packages for each user with a custom logic. We are planning to do this using EF Core's dynamic filter feature.

So, basically we want to use BaGet as a nuget package. We don't want to modify the source code of BaGet and want to upgrade when you release a new version on nuget.org easily.

We can do this by adding BaGet's source code to our project if you don't have a plan to add it to nuget.org.

Thanks again for your great work 👍

**Comment by loic-sharma** [comment]
Interesting! Unfortunately, BaGet isn't in a great place to be extended like this. My recommendations:

1. Copy the source code in [this folder](https://github.com/loic-sharma/BaGet/tree/2772c1e29a82cefe914dec1b6e3b89028a0a3d76/src/BaGet)
1. Replace [these `ProjectReference`s](https://github.com/loic-sharma/BaGet/blob/2772c1e29a82cefe914dec1b6e3b89028a0a3d76/src/BaGet/BaGet.csproj#L25) into `PackageReference`s
1. Add whatever changes you need

This way, you'll copy as little code as possible. Do note that BaGet is in "prerelease" mode and doesn't have stable API surface yet. This is something I'd like to add in the future, but there's too much churn right now.

**Comment by ismcagdas** [comment]
@loic-sharma thanks a lot, actually I was looking for BaGet.* packages and it seems like they all are on nuget. Thanks again 😄
